src/main/java/com/coveo/nashorn_modules/Module.java

```diff
@@ -131,7 +132,11 @@ public Object require(String module) throws ScriptException {
 
     } finally {
       // Finally, remove successful resolves from the refCache
-      refCache.remove();
+      if(requestedFullPath != null){
+        refCache.get().remove(requestedFullPath);
+      }else{
+        refCache.remove();;
```

Comment:
I figured, perhaps incorrectly, that there are use cases (as yet without tests) for which it would make sense to maintain the previous behaviour and clean the ref-cache. So requestedFullPath was not resolved (which is OK sometimes I think?), but the ref-cache might still have entries.

So perhaps I should remove this (or try to add a test)?